quickwit/quickwit-indexing/src/models/indexing_directory.rs

```diff
@@ -1,179 +0,0 @@
-    }
-}
-
-impl IndexingDirectory {
-    pub async fn create_in_dir<P: AsRef<Path>>(dir_path: P) -> anyhow::Result<IndexingDirectory> {
-        let root_dir = dir_path.as_ref().to_path_buf();
-
-        // Delete if exists and recreate scratch directory.
-        let scratch_directory_path = root_dir.join(SCRATCH);
-        fs::create_dir_all(&scratch_directory_path).await?;
-
-        ignore_error_kind!(
-            io::ErrorKind::NotFound,
-            fs::remove_dir_all(&scratch_directory_path).await
-        )
-        .with_context(|| {
-            format!(
-                "Failed to empty scratch directory `{}`.",
-                scratch_directory_path.display(),
-            )
-        })?;
-        fs::create_dir(&scratch_directory_path)
-            .await
-            .with_context(|| {
-                format!(
-                    "Failed to create scratch directory `{}`. ",
-                    scratch_directory_path.display(),
-                )
-            })?;
-        let scratch_directory = ScratchDirectory::new_in_dir(scratch_directory_path);
-        let inner = InnerIndexingDirectory {
-            root: Root::Dir(root_dir),
-            scratch_directory,
-        };
-        let indexing_directory = IndexingDirectory {
-            inner: Arc::new(inner),
-        };
-        Ok(indexing_directory)
-    }
-
-    pub fn scratch_directory(&self) -> &ScratchDirectory {
-        &self.inner.scratch_directory
-    }
-
-    pub fn path(&self) -> &Path {
-        match &self.inner.root {
-            Root::Dir(root) => root,
-            #[cfg(any(test, feature = "testsuite"))]
-            Root::TempDir(tempdir) => tempdir.path(),
-        }
-    }
-
-    #[cfg(any(test, feature = "testsuite"))]
-    pub async fn for_test() -> Self {
-        let tempdir = tempfile::tempdir().unwrap();
-
-        let scratch_directory_path = tempdir.path().join(SCRATCH);
-        fs::create_dir_all(&scratch_directory_path).await.unwrap();
-
-        let scratch_directory = ScratchDirectory::new_in_dir(scratch_directory_path);
-
-        let inner = InnerIndexingDirectory {
-            root: Root::TempDir(tempdir),
-            scratch_directory,
-        };
-        IndexingDirectory {
-            inner: Arc::new(inner),
-        }
-    }
-
-    pub fn downgrade(&self) -> WeakIndexingDirectory {
-        WeakIndexingDirectory {
-            inner: Arc::downgrade(&self.inner),
-        }
-    }
-}
-
-#[cfg(test)]
-mod tests {
-    use super::*;
-
-    #[tokio::test]
-    async fn test_indexing_directory() -> anyhow::Result<()> {
-        let tempdir = tempfile::tempdir()?;
-        let indexing_directory = IndexingDirectory::create_in_dir(tempdir.path()).await?;
-        let indexing_directory_path = indexing_directory.path().to_path_buf();
-        assert_eq!(indexing_directory_path, tempdir.path());
-
-        let scratch_directory_path = indexing_directory_path.join("scratch");
-        assert!(scratch_directory_path.try_exists()?);
```

Comment:
We lost this unit test. Can you port it?

Reply:
Done!